velodyne_driver/src/lib/input.cc

```diff
@@ -199,11 +199,9 @@ namespace velodyne_driver
                          << nbytes << " bytes");
       }
 
-    // Average the times at which we begin and end reading.  Use that to
-    // estimate when the scan occurred. Add the time offset.
-    double time2 = ros::Time::now().toSec();
-    pkt->stamp = ros::Time((time2 + time1) / 2.0 + time_offset);
-
+    // Use ros time now as when we start getting packet as the time of scan. At this point
```

Comment:
my brain hurts a bit from this sentence. Should it be:
"Use ros time now _from_ when we _started_ getting a packet as the time of _a_ scan"

Also, one scan contains many packets. This implies that one of them sets the timestamp for the whole scan, an explanation what happens downstream with the time would be nice.

Reply:
fixed.